SofaKernel/framework/sofa/core/topology/TopologyHandler.cpp

```diff
@@ -48,6 +48,8 @@ void TopologyHandler::ApplyTopologyChanges(const std::list<const core::topology:
     for (changeIt=_changeList.begin(); changeIt!=_changeList.end(); ++changeIt)
     {
         core::topology::TopologyChangeType changeType = (*changeIt)->getChangeType();
+        std::string topoChangeType = "DefaultTopologyHandler: " + parseTopologyChangeTypeToString(changeType);
```

Comment:
I would not put string allocation + reallocation in a loop.

Reply:
It is already what is doing parseTopologyChangeTypeToString. But the first part is to know in which class we are.

Reply:
Why not precomputing "DefaultTyopologyHandler+xxx" instead of re-computing it a loop ?

Reply:
Im not sure to understand, you mean having inside the class DefaultTopologyHandler a new map:
```
map<core::topology::TopologyChangeType, std::string> handlerMap;
handlerMap[changeType] = " DefaultTopologyHandler: " + parseTopologyChangeTypeToString(changeType);
```

Reply:
Yes, it may even be a static map in the function.